source/core/src/main/com/csse3200/game/components/gamearea/GameAreaDisplay.java

```diff
@@ -1,44 +1,316 @@
 package com.csse3200.game.components.gamearea;
 
 import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.graphics.g2d.BitmapFont;
 import com.badlogic.gdx.graphics.g2d.SpriteBatch;
-import com.badlogic.gdx.scenes.scene2d.ui.Label;
+import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.scenes.scene2d.Actor;
+import com.badlogic.gdx.scenes.scene2d.InputEvent;
+import com.badlogic.gdx.scenes.scene2d.InputListener;
+import com.badlogic.gdx.scenes.scene2d.Stage;
+import com.badlogic.gdx.scenes.scene2d.ui.*;
+import com.badlogic.gdx.scenes.scene2d.utils.ChangeListener;
+import com.badlogic.gdx.scenes.scene2d.utils.DragAndDrop;
+import com.badlogic.gdx.scenes.scene2d.utils.Drawable;
+import com.badlogic.gdx.scenes.scene2d.utils.TextureRegionDrawable;
+import com.badlogic.gdx.utils.Null;
+import com.csse3200.game.components.maingame.MainGameExitDisplay;
 import com.csse3200.game.ui.UIComponent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.badlogic.gdx.scenes.scene2d.ui.Dialog;
+import com.badlogic.gdx.scenes.scene2d.ui.Table;
+
+import java.util.concurrent.TimeUnit;
 
 /**
  * Displays the name of the current game area.
  */
 public class GameAreaDisplay extends UIComponent {
+  private static final Logger logger = LoggerFactory.getLogger(GameAreaDisplay.class);
+  private static final float Z_INDEX = 2f;
+  // Dialog for displaying tower details
+
+  private Vector2[] towerPositions1;
```

Comment:
It's not clear what is happening in this method. What are all the tower variables here for?

Reply:
this is just a structural design for our functionality of drag and drop, we will integrate it with the real entities for the upcoming sprint